packages/store/src/keys/compareKeys.js

```diff
@@ -99,11 +101,12 @@ export const compareKeys = (left, right) => {
         }
       }
       // If copyRecord X is smaller than copyRecord Y, then they must have the
-      // same property names, and every value is X must be smaller or equal to
-      // every corresponding value in Y. The rank order of X and Y is based
-      // on lexicographic rank order of their values, as organized by the
-      // reverse order of their property names. Thus
-      // if compareKeys(X,Y) < 0 then compareRank(X,Y) <= 0.
+      // same property names and every value in X must be smaller or equal to
+      // the corresponding value in Y (with at least one value smaller).
+      // The rank order of X and Y is based on lexicographic rank order of
+      // their values, as organized by reverse lexicographic order of their
+      // property names.
+      // Thus if compareKeys(X,Y) < 0 then compareRank(X,Y) < 0.
```

Comment:
Correspondingly, at least one value of the two records differs and therefore a ranking based on values should not place them in the same equivalence class.

Reply:
Good catch! This is true and stronger than what I intended to claim at the time. The reason is a really cool invariant about the way that `compareKeys` can be more precise than `compareRank`:

if `compareRank(X,Y) === 0` then `compareKeys(X,Y)` is either `0` or `NaN`.